src/qibo/gates/gates.py

```diff
@@ -2535,9 +2535,12 @@ def __init__(
         super().__init__(trainable)
         self.name = "grbs"
         self.draw_label = "gRBS"
-        self.target_qubits = tuple(qubits_in) + tuple(qubits_out)
         self.unitary = True
 
+        target_qubits = list(qubits_in) + list(qubits_out)
+        target_qubits.sort()
+        self.target_qubits = tuple(target_qubits)
```

Comment:
Looking at this and following our previous discussion, I initially struggled to understand how it works, because here it seems like you are losing all information about the user-passed qubit order, for example `GeneralizedRBS([0, 1], [2])` will end up having the same `target_qubits` with `GeneralizedRBS([0, 2], [1])` which normally shouldn't be the case. However, after doing some doing some quick tests locally, it seemed that this fix works well and the "cheat" actually happens in https://github.com/qiboteam/qibo/blob/f1a7574577647a0e683b44372f6a501351b8e38e/src/qibo/backends/numpy.py#L132-L138
where you are retaining which are the `qubits_in` and `qubits_out`.

I am not sure if there is a cleaner solution, but I guess that should be fine for now.